### ISSUE TITLE
Rewrite WPA handshake handling

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#2079]
# Issue type

- Defect - Incorrect value displayed/received/stored
- Defect - Unexpected behavior (obvious or confirmed in the forum)

# Aircrack-ng version

- Version: 1.5.2
- Commit Revision hash: 672c053

# Defect

- Rewrite the handshake detection (as a library), and favor complete handshake over partial ones.
- It also need to handle the WPA3 handshake, which requires more than the eapol packets.
- In Airodump-ng, export individual handshakes in separate pcap files as well (if pcap output format is selected).
- Aircrack-ng needs to select the best handshake possible, actually count the amount of usable handshakes, and export correctly the hccapx
- Improve [Tutorial: WPA Packet Capture Explained](https://aircrack-ng.org/doku.php?id=wpa_capture) 

## Related issues

There are quite a few but the most recent is #2077. Here is the search string: https://github.com/aircrack-ng/aircrack-ng/issues?q=is%3Aissue+is%3Aopen+handshake

**Comment by ZerBea** [comment]
Good idea. Maybe you can add EAPOL time measurement (inclusive correct time stamp) and nonce error detection/correction, too. That will be a big improvement!
Please add wpaclean also to labels, because we have similar issues here, too.
